File: backend/ResearchCruiseApp/Application/Models/DTOs/CruiseApplications/CruiseApplicationDto.cs
namespace ResearchCruiseApp.Application.Models.DTOs.CruiseApplications;

public class CruiseApplicationDto
{
    public Guid Id { get; init; }

    public string Number { get; init; } = null!;

    public DateOnly Date { get; init; }

    public int Year { get; init; }

    public Guid CruiseManagerId { get; init; }

    public string CruiseManagerEmail { get; set; } = null!;

    public string CruiseManagerFirstName { get; set; } = null!;

    public string CruiseManagerLastName { get; set; } = null!;

    public Guid DeputyManagerId { get; init; }

    public string DeputyManagerEmail { get; set; } = null!;

    public string DeputyManagerFirstName { get; set; } = null!;

    public string DeputyManagerLastName { get; set; } = null!;

    public bool HasFormA { get; init; }

    public bool HasFormB { get; init; }

    public bool HasFormC { get; init; }

    public int Points { get; set; }

    public string Status { get; init; } = null!;

    public string EffectsDoneRate { get; set; } = "0";

    public string? Note { get; init; }

    public string? CruiseHours { get; init; }

    public int? CruiseDays { get; set; }

    public string? AcceptablePeriodBeg { get; init; }

    public string? AcceptablePeriodEnd { get; init; }

    public string? OptimalPeriodBeg { get; init; }

    public string? OptimalPeriodEnd { get; init; }

    public DateTime? PrecisePeriodStart { get; init; }

    public DateTime? PrecisePeriodEnd { get; init; }

    public DateTime? StartDate { get; init; }

    public DateTime? EndDate { get; init; }
}

File: frontend/src/api/applications/dto/CruiseApplicationDto.ts
export enum CruiseApplicationStatus {
  Draft = 'draft',
  Accepted = 'accepted',
  WaitingForSupervisor = 'waitingForSupervisor',
  AcceptedBySupervisor = 'acceptedBySupervisor',
  Denied = 'denied',
  DeniedBySupervisor = 'deniedBySupervisor',
  FormBRequired = 'formBRequired',
  FormBFilled = 'formBFilled',
  Undertaken = 'undertaken',
  Reported = 'reported',
}

export type CruiseApplicationDto = {
  id: string;
  number: string;
  year: number;
  date: string;

  cruiseManagerId: string;
  cruiseManagerEmail: string;
  cruiseManagerFirstName: string;
  cruiseManagerLastName: string;

  deputyManagerId: string;
  deputyManagerEmail: string;
  deputyManagerFirstName: string;
  deputyManagerLastName: string;

  hasFormA: boolean;
  hasFormB: boolean;
  hasFormC: boolean;

  points: number;
  status: CruiseApplicationStatus;
  effectsDoneRate: string;
  note: string | null;

  cruiseHours: string;
  cruiseDays: number | null;
  acceptablePeriodBeg: string | null;
  acceptablePeriodEnd: string | null;
  optimalPeriodBeg: string | null;
  optimalPeriodEnd: string | null;
  precisePeriodStart: string | null;
  precisePeriodEnd: string | null;
  startDate: string | null;
  endDate: string | null;
};
